src/lsrs/web/utility/mapping-helper.ts

```diff
@@ -0,0 +1,101 @@
+import {NextFunction, Request, Response} from "express";
+import {ResponseWrapper} from "../model/common";
+
+/**
+ * Abstract base implementation providing a convenient way to register an endpoint in Express.
+ * Calling the register method of the actual implementations creates an arrow expression, expecting the standard Express
+ * request handler parameters (request, response and next function) for the given controller entry point. Since the
+ * controllers might expect a specific request model, this helper component also deals with converting the "raw" request
+ * object into the defined request model. Also attaches a response handler and an exception handler to each of the registrations.
+ */
+abstract class MappingHelper {
+
+    /**
+     * Converts the response of the controller into an Express response object, by doing the following steps:
+     *  - Adds the defined headers;
+     *  - Sets the response status;
+     *  - And sends the response body (by either sending the raw response content directly, or converting it to JSON).
+     *
+     * @param responseWrapper ResponseWrapper object containing the controller response context
+     * @param response Express Response object
+     */
+    protected mapResponse(responseWrapper: ResponseWrapper<any>, response: Response): void {
+
+        responseWrapper.headers
+            .forEach((value, key) => response.setHeader(key, value))
+
+        response.status(responseWrapper.status);
+
+        if (responseWrapper.content) {
+            if (responseWrapper.sendAsRaw) {
+                response.send(responseWrapper.content);
+            } else {
+                response.json(responseWrapper.content);
+            }
+        } else {
+            response.send();
+        }
+    }
+
+    /**
+     * Handles errors thrown by the attached controller endpoint by passing it forward in next Express function.
+     *
+     * @param endpointCall attached raw endpoint call
+     */
+    protected catchAsyncError(endpointCall: (request: Request, response: Response) => void): (request: Request, response: Response, next: NextFunction) => void {
+
+        return async (request, response, next) => {
+
+            try {
+                await endpointCall(request, response);
+            } catch (error) {
+                next(error);
+            }
+        };
+    }
+}
+
+/**
+ * MappingHelper implementation for endpoint registrations not expecting any request model.
+ */
+export class ParameterlessMappingHelper extends MappingHelper {
+
+    /**
+     * Registers the given endpoint without request model conversion.
+     * Attaches response mapping and async error handling.
+     *
+     * @param endpointCall controller endpoint call to be registered
+     * @returns Express request handler definition
+     */
+    public register(endpointCall: () => Promise<ResponseWrapper<any>>): (request: Request, response: Response, next: NextFunction) => void {
```

Comment:
Would it make sense to spec the parameter lambda return type as `Promise<ResponseWrapper<any>> | ResponseWrapper<any>`? I think you can await it either way, and that way you wouldn't force using async methods even if they don't need to await anything.

Reply:
Recursive reference back to your first comment. :) Will check it, but I believe it was raising a syntax error, when I didn't define the Promise<> generic type in the specification.

Reply:
Yes, it totally made sense, thanks! It also let me revert the actuator endpoint changes.